prototype/core/impl/src/main/java/org/eclipse/sensinact/prototype/model/nexus/impl/ModelNexus.java

```diff
@@ -660,36 +648,26 @@ private EReference getServiceForModel(EClass model, String serviceName) {
         return feature == null ? null : (EReference) feature;
     }
 
-    public Stream<EAttribute> getResourcesForService(EClass svcClass) {
-        return svcClass.getEAllAttributes().stream().filter(a -> {
-            EAnnotation annotation = a.getEAnnotation("metadata");
-            if (annotation == null) {
-                // TODO do we have to special case admin, or should it have a metadata
-                // annotation
-                return sensinactPackage.getAdmin().equals(a.getEContainingClass());
-            } else {
-                return annotation.eContents().stream().anyMatch(ModelMetadata.class::isInstance);
-            }
-        });
+    public Stream<ETypedElement> getResourcesForService(EClass svcClass) {
+        List<ETypedElement> result = new ArrayList<>(svcClass.getEAllAttributes());
+        result.addAll(svcClass.getEOperations());
+        return result.stream();
```

Comment:
Stream.concat(svcClass.getEAllAttributes().stream(), svcClass.getEOperations().stream()) would avoid creating an intermediate list.

Reply:
Thanks, I know something like this should exists, but I couldn't find it.

Reply:
should we sort this concatinated stream automatically sort by name?

Reply:
We shouldn’t sort it - consumers can sort according to their needs, but in many cases no sorting will be required.

Reply:
Also - should this be getEAllOperations()